[ProvisioningAPI] Allow specifying group display name during creation

m7913d opened the pull request:
Signed-off-by: Dries Mys <dries.mys@my-dreams.be>

m7913d committed:
[ProvisioningAPI] Allow specifying group display name during creation

Signed-off-by: Dries Mys <dries.mys@my-dreams.be>

m7913d committed:
Fixed indentation

Signed-off-by: Dries Mys <dries.mys@my-dreams.be>

m7913d commented:
The failed check seems to be unrelated to this commit. Should I do something to continue the PR?

kesselb commented:
> @kesselb The failed check (continuous-integration/drone/pr) seems to be unrelated to this commit. Should I do something to continue the PR? Note that my first commit run this check successful and the second commit only fixed the indentation.

Thanks for your pull request :+1: Please ignore the ci failure.

skjnldsv commented on apps/provisioning_api/lib/Controller/GroupsController.php:
```diff
@@ -247,7 +248,10 @@ public function addGroup(string $groupid): DataResponse {
 		}
-		$this->groupManager->createGroup($groupid);
+		$group = $this->groupManager->createGroup($groupid);
+		if ($displayname !== '') {
```
Please also check if not null

skjnldsv replied:
I'm just repeating what psalm is saying :wink: :parrot:

m7913d replied:
@blizzz I added a null check and I throw an exception if it is the case. However, this results in a failure of the nodb test (see continuous integration). Shouldn't I throw an exception in this case or is the failing nodb test wrong? It seems to me appropriate that an error message is returned in case the group creation failed.

blizzz commented:
I would prefer having integration tests here, extending on https://github.com/nextcloud/server/blob/master/build/integration/features/provisioning-v1.feature#L199-L217

m7913d committed:
Group may be null if backend doesn't support creating groups

Signed-off-by: Dries Mys <dries.mys@my-dreams.be>

m7913d committed:
Added tests for specifying custom display name during group creation.

Signed-off-by: Dries Mys <dries.mys@my-dreams.be>

m7913d commented:
> I would prefer having integration tests here, extending on https://github.com/nextcloud/server/blob/master/build/integration/features/provisioning-v1.feature#L199-L217

I added some test: see my last commit. Do I need to add other test cases?

blizzz commented:
CI is unhappy

m7913d commented:
> CI is unhappy

I know, a nodb test is failing, because I throw now an exception in case the group creation failed. This seems appropriate to me to report the failure to the api caller. See my remark to @skjnldsv who requested a null check. What  solution do you suggest? Not throwing or changing the failing nodb test?

m7913d committed:
Effectively return a valid group as it would be the case in a real
situation. Returned group object is now used to set the displayname if
provided.

Signed-off-by: Dries Mys <dries.mys@my-dreams.be>

m7913d commented:
@blizzz @skjnldsv I corrected the failed test as the test environment did not return a valid group. The acceptance-user test is still failing, but I assume this is unrelated to this PR as after rebasing this feature branch to the latest master branch, another scenario of this test failed with the following reason: "The user disabledUser in the list of users is not shown yet after 100 seconds".

blizzz commented:
I saw it failing somewhere else, too, let me restart nevertheless

m7913d commented:
@blizzz Thanks for rerunning the failed check. Now all checks have passed/
